src/main/java/rs/teslaris/core/controller/InstitutionController.java

```diff
@@ -0,0 +1,87 @@
+package rs.teslaris.core.controller;
+
+import javax.transaction.Transactional;
+import javax.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import rs.teslaris.core.converter.institution.OrganisationUnitConverter;
+import rs.teslaris.core.dto.institution.OrganisationUnitDTO;
+import rs.teslaris.core.dto.institution.OrganisationUnitDTORequest;
+import rs.teslaris.core.model.commontypes.ApproveStatus;
+import rs.teslaris.core.service.OrganisationUnitService;
+
+@RestController
+@RequestMapping("/api/institution")
+@RequiredArgsConstructor
+public class InstitutionController {
```

Comment:
I think we should rename this controller to OrganisationUnitController, because this controller will be used also for departments and chairs. RequestMapping path should be also changed. It looks to me completely confusing get paths how you defined at the moment. To get an organisation unit I need {app}/api/institution/organisation-units/3656

Reply:
And organisation-units is in all method level mappings, so why it is not moved to class level mapping?

Reply:
Fixed, actually not sure why I named it this way.